test/integration/targets/service_facts/tasks/main.yml

```diff
@@ -0,0 +1,24 @@
+# Test playbook for the service_facts module
+# (c) 2017, Adam Miller <admiller@redhat.com>
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+- name: Gather service facts
+  service_facts:
+
+- name: check for ansible_facts.services exists
+  assert:
+    that: ansible_facts.packages is defined
```

Comment:
Wonder if we could check the size of the list, rather than it just being defined (and possibly empty)?

Reply:
it should be a dict, not a list